xds/internal/resolver/serviceconfig.go

```diff
@@ -53,28 +57,127 @@ type cdsBalancerConfig struct {
 	Cluster string `json:"cluster"`
 }
 
-func serviceUpdateToJSON(su xdsclient.ServiceUpdate) (string, error) {
+type route struct {
+	Path     *string                    `json:"path,omitempty"`
+	Prefix   *string                    `json:"prefix,omitempty"`
+	Regex    *string                    `json:"regex,omitempty"`
```

Comment:
Do these need to be pointers. I believe that the `omitempty` tag handles empty strings properly.

Reply:
That sounds right.

Keeping them as pointers makes the assign at line 84 easier though. (No need to check for nil)